src/aat/java/uk/gov/hmcts/ccd/fta/steps/BackEndFunctionalTestScenarioPlayer.java

```diff
@@ -0,0 +1,270 @@
+                    Assert.fail("Dynamic value for request query parameter '" + queryParam + "' does not exist");
+                } else {
+                    aRequest.queryParam(queryParam, value);
+                }
+            });
+        }
+
+        if (requestData.getBody() != null) {
+            aRequest.body(new ObjectMapper().writeValueAsBytes(requestData.getBody()));
+        }
+
+        scenarioContext.setTheRequest(aRequest);
+        scenario.write(JsonUtils.getPrettyJsonFromObject(scenarioContext.getTestData().getRequest()));
+    }
+
+    @Override
+    @When("it is submitted to call the [{}] operation of [{}]")
+    public void submitTheRequestToCallAnOperationOfAProduct(String operation, String productName) throws IOException {
+        boolean isCorrectOperation = scenarioContext.getTestData().meetsOperationOfProduct(operation, productName);
+        String errorMessage = "Test data does not confirm it is calling the following operation of a product: "
+            + operation + " -> " + productName;
+        Assert.assertTrue(errorMessage, isCorrectOperation);
+
+        RequestSpecification theRequest = scenarioContext.getTheRequest();
+        String uri = scenarioContext.getTestData().getUri();
+
+        Response response = null;
+        Assert.assertNotNull("No request method in data file", scenarioContext.getTestData().getMethod());
+        switch (scenarioContext.getTestData().getMethod()) {
+            case "GET":
+                response = theRequest.get(uri);
+                break;
+            case "POST":
+                response = theRequest.post(uri);
+                break;
+            case "PUT":
+                response = theRequest.put(uri);
+                break;
+            case "DELETE":
+                response = theRequest.delete(uri);
+                break;
+            default:
+                Assert.fail("Unknown request method in data file");
+        }
+
+        QueryableRequestSpecification queryableRequest = SpecificationQuerier.query(theRequest);
+        scenario.write(queryableRequest.getMethod() + " " + queryableRequest.getURI());
+
+        Map<String, Object> responseHeaders = new HashMap<>();
+        response.getHeaders().forEach(header -> responseHeaders.put(header.getName(), header.getValue()));
+        ResponseData responseData = new ResponseData();
+        responseData.setResponseCode(response.getStatusCode());
+        responseData.setResponseMessage(HttpStatus.valueOf(response.getStatusCode()).getReasonPhrase());
+        responseData.setHeaders(responseHeaders);
+
+        if (!response.getBody().asString().isEmpty()) {
+            responseData.setBody(JsonUtils.readObjectFromJsonText(response.getBody().asString(), Map.class));
+        }
+
+        scenarioContext.setTheResponse(responseData);
+    }
+
+    @Override
+    @Then("a positive response is received")
+    public void verifyThatAPositiveResponseWasReceived() {
```

Comment:
we can merge this step with the one below since it's a bit of duplication accept for minor error message detail - the step name would speak for itself when it fails anyway. You could parameterise the negative/positive word in the step into regex or use the {word} cucumber expression and parse it into the error message if you wanted to keep a custom error message

Alternatively have a look at the new 'Repeatable annotations' feature that just came out in cucumber-jvm 5!

Reply:
Not too sure what you mean by 2 elements of menu choices. Still would be nice to use repeatable annotations here like below but it's purely aesthetic change 

`  @Override
    @Then("a positive response is received")
    @Then("a negative response is received")
    public void verifyCorrectResponseWasReceived(){
        //code
    }`